### PR TITLE
add and update blobber separated

### PR DESCRIPTION
**Pull request by Hitenjain14**
## Fixes
addBlobber() could update blobber too. 
Separated the functionality of both , if any repos are using addBlobber() to update blobber , they should use blobber_update_settings .

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber: 
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

File: code/go/0chain.net/smartcontract/storagesc/blobber_test.go [hunk 1]
```diff
@@ -1322,3 +1298,26 @@ func TestBlobberHealthCheck(t *testing.T) {
...
+
+	var (
+		blob   = addBlobber(t, ssc, 2*GB, tp, avgTerms, 50*x10, balances)
+		b, err = ssc.getBlobber(blob.id, balances)
+	)
+	require.NoError(t, err)
+
+	b.BaseURL = "https://newabcurl.com"
+	_, err = updateBlobberAdd(t, b, 0, tp, ssc, balances)
+	require.Error(t, err)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/blobber_test.go:1323]: Please assert the error message to make sure it returns the expected error.

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Comment by peterlimg** [comment]
@Hitenjain14 please do search the add blobber endpoint in https://github.com/0chain/blobber repo to see if it is used for updating, and correct it if any.

**Comment by Hitenjain14** [comment]
> @Hitenjain14 please do search the add blobber endpoint in https://github.com/0chain/blobber repo to see if it is used for updating, and correct it if any.

I have raised this  [pr](https://github.com/0chain/blobber/pull/961) to address this issue , talked with @dabasov and we shouldn't be doing any updates from blobber server. Should only be done from cli and thats sending `update_blobber_settings` transaction.

File: code/go/0chain.net/smartcontract/storagesc/helper_test.go [hunk 2]
```diff
@@ -171,6 +172,18 @@ func updateBlobber(t testing.TB, blob *StorageNode, value currency.Coin, now int
 	return ssc.addBlobber(tx, input, balances)
 }
 
+func updateBlobber(t testing.TB, blob *StorageNode, value currency.Coin, now int64,
```
Comment (dabasov) [inline code/go/0chain.net/smartcontract/storagesc/helper_test.go:175]: looks like it is duplicate of `updateBlobberAdd`

**Review by dabasov — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/helper_test.go (hunk 2 above)
Comment (Hitenjain14) [reply code/go/0chain.net/smartcontract/storagesc/helper_test.go:175]: Yes but in `updateBlobberAdd` we are calling `ssc.addBlobber(tx, input, balances)` , its used to test that we cannot perform any updates using `addBlobber` now

File: code/go/0chain.net/smartcontract/storagesc/stakepool.go [hunk 3]
```diff
@@ -361,6 +361,28 @@ func (ssc *StorageSmartContract) getOrCreateStakePool(
 	return sp, nil
 }
 
+func (ssc *StorageSmartContract) createStakePool(
+	conf *Config,
+	providerType spenum.Provider,
+	providerId datastore.Key,
+	settings stakepool.Settings,
+	balances chainstate.StateContextI,
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/stakepool.go]: providerType, providerId and balances are not being used, let's remove them

**Review by peterlimg — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/smartcontract/storagesc/helper_test.go (hunk 2 above)
Comment (peterlimg) [reply code/go/0chain.net/smartcontract/storagesc/helper_test.go:175]: @Hitenjain14 if `updateBlobberAdd` would confuse users, let's rename it to `addBlobber`.

Comment (Hitenjain14) [reply code/go/0chain.net/smartcontract/storagesc/helper_test.go:175]: Its a helper test function , using `addblobber` to update the blobber which should fail and thats what we check .

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | blobber-add-update  |
| blobber  | blobber-server-update  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Review by dabasov — APPROVED** [review]
lgtm

**Pull request merged**